src/Evolution/Systems/GrMhd/ValenciaDivClean/BoundaryCorrections/Hll.cpp

```diff
@@ -109,63 +139,44 @@ double Hll::dg_package_data(
       Scalar<DataVector> sound_speed_squared =
           equation_of_state.sound_speed_squared_from_density_and_temperature(
               rest_mass_density, temperature, electron_fraction);
+      get(sound_speed_squared) = clamp(get(sound_speed_squared), 0.0, 1.0);
 
-      // Get v0 from Font
-      Scalar<DataVector> v0 = tilde_d;
-      dot_product(make_not_null(&v0), spatial_velocity, normal_covector);
-      get(v0) *= get(lapse);
+      // Compute v_dot_normal, v^i n_i
+      dot_product(make_not_null(&v_dot_normal), spatial_velocity,
+                  normal_covector);
+      get(v_dot_normal) *= get(lapse);
```

Comment:
I think our two original formulae are the same, but you are having additional lapse factors by multiplying v_dot_normal here (giving you my v0), then still multiplying, but using this as v_n in the discriminant and still multiplying by the lapse in the final expression. In your formulation, I think you don't want L147. Then you should get an expression equivalent to mine, if I am not mistaken.

Reply:
You're right, I've removed this!